src/care/CUDAWatchpoint.h

```diff
@@ -9,15 +9,25 @@
 #define _CARE_CUDA_WATCHPOINT_H_
 
 #ifdef CARE_DEBUG
-#if __CUDACC__
+#if defined(__GPUCC__)
 
+#if __CUDACC__
 // Other library headers
 #include "cuda.h"
+#define care_watchpoint_err_t cudaError_t
+#else
+#include "hip/hip_runtime.h"
+#define care_watchpoint_err_t hipError_t
+#endif
 
-inline void watchpoint_gpuAssert(cudaError_t code, const char *file, int line, bool abort=true)
+inline void watchpoint_gpuAssert(care_watchpoint_err_t code, const char *file, int line, bool abort=true)
 {
    if (code != cudaSuccess) {
+#if __CUDACC__
```

Comment:
`defined(__CUDACC__)`

Reply:
There's a couple more after this that also need to be fixed.

Reply:
fixed